library/src/main/java/com/parse/FacebookController.java

```diff
@@ -131,11 +136,37 @@ public void onError(FacebookException e) {
     return tcs.getTask();
   }
 
+  /**
+   * Get auth data from the access token.
+   * Includes the following:
+   * - UserId
+   * - Access Token
+   * - Expiration Date
+   * - Last Refresh Date
+   * - Permissions (Comma Delineated)
+   * @param accessToken - Facebook's {@link AccessToken}
+   * @return - {@link Map} of auth data used parse to create facebook {@link AccessToken} by hand.
+   * See {@link FacebookController#setAuthData(Map)}
+   */
   public Map<String, String> getAuthData(AccessToken accessToken) {
     Map<String, String> authData = new HashMap<>();
     authData.put(KEY_USER_ID, accessToken.getUserId());
     authData.put(KEY_ACCESS_TOKEN, accessToken.getToken());
     authData.put(KEY_EXPIRATION_DATE, PRECISE_DATE_FORMAT.format(accessToken.getExpires()));
+    authData.put(KEY_REFRESH_DATE, PRECISE_DATE_FORMAT.format(accessToken.getLastRefresh()));
+
+    Set<String> permissionSet = accessToken.getPermissions();
+    StringBuilder stringBuilder = new StringBuilder();
+    for (String permission : permissionSet) {
+      stringBuilder.append(permission);
+      stringBuilder.append(",");
+    }
+    if (stringBuilder.length() > 0) {
```

Comment:
Can you use TextUtils.json(https://developer.android.com/reference/android/text/TextUtils.html#join(java.lang.CharSequence, java.lang.Iterable)?

https://android.googlesource.com/platform/frameworks/base/+/b2c4f0bf11f38fd31d80f1256c89b9db043a2929/core/java/android/text/TextUtils.java